src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/task/VisitTask.kt

```diff
@@ -5,40 +5,31 @@ import org.slf4j.LoggerFactory
 import org.springframework.beans.factory.annotation.Value
 import org.springframework.scheduling.annotation.Scheduled
 import org.springframework.stereotype.Component
-import uk.gov.justice.digital.hmpps.visitscheduler.model.VisitFilter
-import uk.gov.justice.digital.hmpps.visitscheduler.model.VisitStatus.RESERVED
 import uk.gov.justice.digital.hmpps.visitscheduler.service.VisitService
-import java.time.LocalDateTime
 
 @Component
 class VisitTask(
   private val visitService: VisitService,
   @Value("\${task.expired-visit.enabled:false}") private val enabled: Boolean,
-  @Value("\${task.expired-visit.validity-minutes:20}") private val expiredPeriod: Long
+  @Value("\${task.expired-visit.validity-minutes:20}") private val expiredPeriod: Int
 ) {
 
+  companion object {
+    val log: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
   @Scheduled(cron = "\${task.expired-visit.cron:0 0/15 * * * ?}")
   fun deleteExpiredReservations() {
     if (!enabled) {
       return
     }
     log.debug("Entered deleteExpiredReservations expiredPeriod : $expiredPeriod")
 
-    val expired = visitService.findVisitsByFilterPageableDescending(
-      VisitFilter(
-        visitStatus = RESERVED,
-        modifyTimestamp = LocalDateTime.now().minusMinutes(expiredPeriod)
-      )
-    ).content
+    val expiredApplicationReferences = visitService.findExpiredApplicationReferences(expiredPeriod)
 
-    if (expired.isNotEmpty()) {
-      log.debug("Expired visits: ${expired.count()}")
+    if (expiredApplicationReferences.isNotEmpty()) {
+      log.debug("Expired visits: ${expiredApplicationReferences.count()}")
```

Comment:
do we need this logged when count is 0?

Reply:
hmm I left it as it was, There is a log.debug("Entered deleteExpiredReservations expiredPeriod : $expiredPeriod")
above, so you could deduce it zero if no debug, but will move it up a line because it is clearer!